common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -551,44 +551,23 @@ private boolean runShutdownHooks() {
             return terminationFuture();
         }
 
-        boolean inEventLoop = inEventLoop();
-        boolean wakeup;
-        int oldState;
+        startThread();
```

Comment:
This is a change in behavior, are you sure this is desired? Previously the state was changed to shutdown before the thread was started, now we unconditionally start the thread before attempting shutdown. Perhaps now the thread will be allowed to process jobs that it previously wouldn't have been allowed to.